src/window.c

```diff
@@ -2566,6 +2772,201 @@ static int sort_function(const void *dest_str_ptr, const void *src_str_ptr)
   return strcasecmp(dest_str, src_str);
 }
 
+static void update_slot_prefix(void)
+{
+  char *world_name;
+  size_t i;
+  size_t pos = 0;
+  int token_pos = -1;
+  char *fmt;
+  boolean parse_token = false;
+  size_t fmt_len;
+  size_t world_len;
+
+  if(cur_slot_prefix != NULL)
+    free(cur_slot_prefix);
+
+  cur_slot_prefix = malloc(sizeof(char) * MAX_PATH);
+
+  fmt = get_config()->save_slots_name;
+  fmt_len = strlen(fmt);
+  for(i = 0; i < fmt_len; i++)
+  {
+    if(fmt[i] == '%')
+    {
+      token_pos = i;
+      break;
+    }
+  }
+
+  // If no tokens were found, just copy the string over as-is.
+  if(token_pos < 0)
+  {
+    strncpy(cur_slot_prefix, fmt, MAX_PATH);
```

Comment:
`strncpy` [has some questionable behavior](http://www.cplusplus.com/reference/cstring/strncpy/), particularly that it's *not guaranteed to null terminate the destination* and zero-fills the unused portion of the destination. The source is shorter than MAX_PATH so the termination "gotcha" isn't an issue at the moment but for future-proofing I tend to lean more towards `snprintf` + add a null for Win32 safety now.

Reply:
Good point. I'll switch that over.

Reply:
Corrected in eec3541 and a03c0ec.